src/main/scala/org/broadinstitute/dsde/workbench/sam/util/LdapSupport.scala

```diff
@@ -60,10 +60,15 @@ trait LdapSupport extends DirectorySubjectNameSupport {
     }
 
   protected def getAttribute(results: Entry, key: String): Option[String] =
-    Option(results.getAttribute(key)).map(_.getValue)
+    Option(results)
+      .map(_.getAttribute(key))
+      .map(_.getValue)
 
   protected def getAttributes(results: Entry, key: String): Set[String] =
-    Option(results.getAttribute(key)).map(_.getValues.toSet).getOrElse(Set.empty)
+    Option(results)
+      .map(_.getAttribute(key))
+      .map(_.getValues.toSet)
+      .getOrElse(Set.empty)
```

Comment:
Putting these in two separate maps in case `_.getAttributes(key)` returns a null. Or is this measure not necessary?

Reply:
this would read better as a for comp:
```
for {
  searchResultEntries <- Option(results)
  attributes <- Option(searchResultEntries.getAttribute(key))
} yield {
  attributes.getValues.toSet
}.getOrElse(Set.empty)
```